src/basic/unit-def.c

```diff
@@ -117,6 +117,13 @@ static const char* const freezer_state_table[_FREEZER_STATE_MAX] = {
 
 DEFINE_STRING_TABLE_LOOKUP(freezer_state, FreezerState);
 
+static const char* const unit_marker_table[_UNIT_MARKER_MAX] = {
+        [UNIT_MARKER_NEEDS_RELOAD]  = "needs-reload",
+        [UNIT_MARKER_NEEDS_RESTART] = "needs-restart",
+};
```

Comment:
hmm, this is a weird table, as it's about a bunch of flags, and thus sparse. this might be ok'ish as long as we only have two flags, but will explode to be a huge sparse table once we add more

Reply:
at the very least add a comment here that when somebody adds more here this must be reworked. i.e. with every bit added the table doubles in size. exponential growth sucks...

Reply:
or maybe we can index the table here via `fsl(UNIT_MARKER_…)`? But not sure we can express that easily in a constant macro

Reply:
Yeah, I should add a comment. I considered this when writing this, but right now we only waste one entry (i.e. 8 bytes), and it just didn't seems important enough. Obviously with many entries this would be bad, but we might not go above a few flags ever.